ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py

```diff
@@ -159,22 +159,26 @@ def _get_p2p_data(self, p2p_link: dict) -> dict:
         node_child_interfaces = get(p2p_link, "port_channel.nodes_child_interfaces")
         # Convert to new data models
         node_child_interfaces = convert_dicts(node_child_interfaces, primary_key="node", secondary_key="interfaces")
-        if member_interfaces := get_item(node_child_interfaces, "node", self.shared_utils.hostname, default={}).get("interfaces"):
+        if (node_data := get_item(node_child_interfaces, "node", self.shared_utils.hostname)) is not None and (
+            member_interfaces := node_data.get("interfaces")
+        ):
             # Port-channel
-            peer_member_interfaces = get_item(
+            portchannel_id = node_data.get("channel_id", int("".join(re.findall(r"\d", member_interfaces[0]))))
```

Comment:
I see you are just moving this code around, IMHO this line pack too much logic. Had to read it several times to understand what it does. Same with peer_member_interfaces. Not a show stopper though.